src/partitioner/io_utils.py

```diff
@@ -0,0 +1,32 @@
+"""Utility functions for reading and writing files"""
+
+import os
+
+import pandas as pd
+from astropy.table import Table
+
+
+def read_dataframe(path="", file_format="parquet") -> pd.DataFrame:
+    """Read a file in as a dataframe"""
+
+    # Perform checks on the provided path
+    if not os.path.exists(path):
+        raise FileNotFoundError(f"File not found at path: {path}")
+    if not os.path.isfile(path):
+        raise FileNotFoundError(
+            f"Directory found at path - requires regular file: {path}"
+        )
+
+    data_frame = pd.DataFrame
+    # Load file using appropriate mechanism
+    if "csv" in file_format:
```

Comment:
Do you want to include all cases where csv is in the name? Or just where csv is the extension? And the same question for the formats below. 

This could be a problem if someone (incorrectly) called a file something like "data_from_fake_parquet.fits"